test/integration/apiserver/flowcontrol/fight_test.go

```diff
@@ -0,0 +1,192 @@
+	ft.countsMutex.Lock()
+	defer ft.countsMutex.Unlock()
+	ft.writeCounts[fs.Name]++
+}
+
+func (ft *fightTest) createController(invert bool, i int) {
+	fieldMgr := fmt.Sprintf("testController%d%v", i, invert)
+	myConfig := rest.CopyConfig(ft.loopbackConfig)
+	myConfig = rest.AddUserAgent(myConfig, fieldMgr)
+	myClientset := clientset.NewForConfigOrDie(myConfig)
+	fcIfc := myClientset.FlowcontrolV1beta1()
+	informerFactory := informers.NewSharedInformerFactory(myClientset, 0)
+	foundToDangling := func(found bool) bool { return !found }
+	if invert {
+		foundToDangling = func(found bool) bool { return found }
+	}
+	ctlr := utilfc.NewTestable(utilfc.TestableConfig{
+		Name:                   fieldMgr,
+		FoundToDangling:        foundToDangling,
+		Clock:                  clock.RealClock{},
+		AsFieldManager:         fieldMgr,
+		InformerFactory:        informerFactory,
+		FlowcontrolClient:      fcIfc,
+		ServerConcurrencyLimit: 200,             // server concurrency limit
+		RequestWaitLimit:       time.Minute / 4, // request wait limit
+		ObsPairGenerator:       metrics.PriorityLevelConcurrencyObserverPairGenerator,
+		QueueSetFactory:        fqtesting.NewNoRestraintFactory(),
+	})
+	ft.ctlrs[invert][i] = ctlr
+	informerFactory.Start(ft.stopCh)
+	go ctlr.Run(ft.stopCh)
+}
+
+func (ft *fightTest) evaluate(tBeforeCreate, tAfterCreate time.Time) {
+	tBeforeLock := time.Now()
+	ft.countsMutex.Lock()
+	defer ft.countsMutex.Unlock()
+	tAfterLock := time.Now()
+	minFightSecs := tBeforeLock.Sub(tAfterCreate).Seconds()
+	maxFightSecs := tAfterLock.Sub(tBeforeCreate).Seconds()
+	minTotalWrites := int(minFightSecs / 10)
+	maxWritesPerWriter := 6 * int(math.Ceil(maxFightSecs/60))
+	maxTotalWrites := (1 + ft.teamSize*2) * maxWritesPerWriter
+	for flowSchemaName, writeCount := range ft.writeCounts {
+		if writeCount < minTotalWrites {
+			ft.t.Errorf("There were a total of %d writes to FlowSchema %s but there should have been at least %d from %s to %s", writeCount, flowSchemaName, minTotalWrites, tAfterCreate, tBeforeLock)
+		} else if writeCount > maxTotalWrites {
+			ft.t.Errorf("There were a total of %d writes to FlowSchema %s but there should have been no more than %d from %s to %s", writeCount, flowSchemaName, maxTotalWrites, tBeforeCreate, tAfterLock)
+		} else {
+			ft.t.Logf("There were a total of %d writes to FlowSchema %s over %v, %v seconds", writeCount, flowSchemaName, minFightSecs, maxFightSecs)
+		}
+	}
+}
+func TestConfigConsumerFight(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.APIPriorityAndFairness, true)()
+	_, loopbackConfig, closeFn := setup(t, 100, 100)
+	defer closeFn()
+	const teamSize = 3
+	ft := newFightTest(t, loopbackConfig, teamSize)
+	tBeforeCreate := time.Now()
+	ft.createMainInformer()
+	ft.foreach(ft.createController)
+	tAfterCreate := time.Now()
+	time.Sleep(110 * time.Second)
+	ft.evaluate(tBeforeCreate, tAfterCreate)
+	close(ft.stopCh)
```

Comment:
nit: make it "defer close(ft.stopCh)" right after creating newFightTest

Reply:
changed in #100061